packages/file_selector/file_selector/example/lib/main.dart

```diff
@@ -0,0 +1,298 @@
+
+  @override
+  _TextDisplayState createState() => _TextDisplayState();
+}
+
+class _TextDisplayState extends State<TextDisplay> {
+  String fileContents;
+
+  @override
+  void initState() {
+    super.initState();
+    _getFileContents();
+  }
+
+  void _getFileContents() async {
+    String contents = await widget.file.readAsString();
+    setState(() => fileContents = contents);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: Text(widget.file.name),
+      content: Scrollbar(
+        child: SingleChildScrollView(
+          child: Text(
+            fileContents ??
+                'Opening file contents...\nThis may take a while if your file is large.',
+          ),
+        ),
+      ),
+      actions: [
+        FlatButton(
+          child: const Text('Close'),
+          onPressed: () {
+            Navigator.pop(context);
+          },
+        ),
+      ],
+    );
+  }
+}
+
+/// Widget that displays a text file in a dialog
+class ImageDisplay extends StatefulWidget {
+  /// File to display
+  final XFile file;
+
+  /// Default Constructor
+  ImageDisplay({Key key, @required this.file}) : super(key: key);
+
+  @override
+  _ImageDisplayState createState() => _ImageDisplayState();
+}
+
+class _ImageDisplayState extends State<ImageDisplay> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: Text(widget.file.name),
+      content: Image.network(widget.file.path),
+      actions: [
+        FlatButton(
+          child: const Text('Close'),
+          onPressed: () {
+            Navigator.pop(context);
+          },
+        ),
+      ],
+    );
+  }
+}
+
+/// Home Page of the application
+class MyHomePage extends StatefulWidget {
+  /// Constructor for MyHomePage
+  MyHomePage({Key key, this.title}) : super(key: key);
+
+  /// Title of Home Page
+  final String title;
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+/// State of Home Page
+class _MyHomePageState extends State<MyHomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            RaisedButton(
+              color: Colors.blue,
+              textColor: Colors.white,
+              child: Text('Press to try saving a file'),
+              onPressed: () => Navigator.pushNamed(context, '/save'),
+            ),
+            SizedBox(height: 10),
+            RaisedButton(
+              color: Colors.blue,
+              textColor: Colors.white,
+              child: Text('Press to try opening a file'),
+              onPressed: () => Navigator.pushNamed(context, '/open'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
```

Comment:
Unless I missed it, neither `openFiles` nor `getDirectoryPath` are exercised/demonstrated in this example.

Reply:
Thanks for the feedback Stuart :) 
I added a new example for `openFiles`. Can we wait until the first non-web implementation to add an example for getDirectoryPath? Please take a look at my other comment.

Reply:
> Can we wait until the first non-web implementation

There are currently zero implementations of this plugin; why would web be the standard that defines what's covered in the main plugin?

I have several desktop implementations essentially complete locally, so it's not like we're talking about a long delay before adding other implementations.

Reply:
Thanks for the feedback Stuart! I already added an example page for getDirectoryPath. With that I think our examples are now covering all the API methods. PTAL

Reply:
Friendly ping!